src/cli/bin.ts

```diff
@@ -164,22 +148,67 @@ yargs
       ) as OutcomeCounters;
       const errorReport = {};
 
-      const fromVersionIndex = fromVersions.indexOf(fromVersion);
-      const toVersionIndex = toVersions.indexOf(toVersion);
+      const doesReportIncludeStacks = stack;
 
-      for (const { migrate } of migrationFunctions.slice(
-        fromVersionIndex,
-        toVersionIndex + 1,
-      )) {
-        await migrate(contentServer, {
-          counters,
+      // Handle the special case of 4.3 to 5.0 separately, as:
```

Comment:
Hmm, I'm not entirely convinced by those two arguments:
- the fact that the content server structure is different doesn't keep us from composing `migrate_43_to_50` with the other migration functions. The logic to parse the legacy content server is inside `migrate_43_to_50` anyway.
- the difference is that we don't use the generic `migrateXyz` functions inside, and that we need the `servers`, which is reflected by the signature of the function being different. But it does not really keep us from composing `migrate_43_to_50` with the other migration functions, right? We could haver `servers` and the `migrateXyz` parameters as optional

So, unless I'm missing something, when I read this comment, I'm not entirely convinced that we **need** to handle this separately from the other migrations. I would adapt the comment, maybe saying more that it's cleaner for `migrate_43_to_50` to have its own signature and to not have mixed with the other functions signatures. Or even not handle the `4.3` case separately, if that's possible. What do you think?

Also, I would add the fact that `migrate_43_to_50` is asynchronous, as opposed to the other ones, because that definitely plays for handling this case separately, right?

Reply:
To sum up:
- The fact that the content server is completely different means that the `getMigrateXyz` functions don't apply.
- The `servers` are needed only in this function and not in the other ones.
- It is the only async one.

These are the arguments for not composing: we don't want to pollute all the other functions by making some arguments optional, or adding unnecessary ones.

I'll add the async argument in the comment 👍 

For the general shape of the comment I'd like it to remain as short and simple as possible. If you have an improvement in mind can you make a suggestion ?

Reply:
I think you could sum up the first two dots with sth like: `its function has a different signature from the other migration functions`. I find that bit more convincing.
I find the sentence "the legacy content server structure is completely different" confusing because each migration step has its own logic anyway, whereas the impact it has (less parameters needed in the signature), is the reason why we're handling it separately.

Feel free to reformulate, or to disagree of course!